libbindgen/src/codegen/mod.rs

```diff
@@ -656,8 +656,30 @@ impl CodeGenerator for CompInfo {
         //
         // TODO: Generate layout tests for template specializations, yay!
         if self.has_non_type_template_params() ||
```

Comment:
You should still bail out if `has_non_type_template_params`, I don't think a lot of stuff will happen, since they probably don't have layout, but still seems a bit clearer.